shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -554,7 +554,16 @@ private void copyFileToLocal(AlluxioURI srcPath, AlluxioURI dstPath)
     File dstFile = new File(dstPath.getPath());
     String randomSuffix =
         String.format(".%s_copyToLocal_", RandomStringUtils.randomAlphanumeric(8));
-    File tmpDst = new File(dstFile.getAbsolutePath() + randomSuffix);
+    File tmpDst;
+    File outputFile;
+    if (dstFile.isDirectory()) {
+      tmpDst = new File(PathUtils.concatPath(dstFile.getAbsolutePath(),
+              srcPath.getName() + randomSuffix));
+      outputFile = new File(PathUtils.concatPath(dstFile.getAbsolutePath(), srcPath));
```

Comment:
oh cool, didn't see that it was already implemented in `AlluxioURI`, good catch :)